spharpy/samplings/coordinates.py

```diff
@@ -310,3 +310,29 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    def _set_weights(self, weights):
+        """Check and set sampling weights.
+
+        Set self._weights, which is an atleast_1d numpy array of shape
+        [L,M,...,N].
+
+        """
+        # check if weights is None
```

Comment:
any special reason why not to all the _weights from Coordinates first?
I mean it setting it before the sum is checked, but its also raising an error so it does not really matter.
https://github.com/pyfar/pyfar/blob/02e8695b714e3ce380522e423c174284c6137020/pyfar/classes/coordinates.py#L2734

Reply:
Same reason as in #121 
If super().method is used, the incorrect weights can be set in the process, which is to be avoided.